glean-core/src/upload/mod.rs

```diff
@@ -0,0 +1,357 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Ping upload helper.
+
+// !IMPORTANT!
+// Remove the next line when this module's
+// functionality is in the Glean object. This
+// is here just to not have lint error for now.
+#![allow(dead_code)]
+
+use std::fs::{remove_dir_all, remove_file, File};
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+
+use lazy_static::lazy_static;
+use log;
+use regex::Regex;
+use serde_json::Value as JsonValue;
+use uuid::Uuid;
+
+use crate::Result;
+use queue::PingsQueue;
+use request::PingRequest;
+
+mod queue;
+mod request;
+
+lazy_static! {
+    /// Ping file names are UUIDs, this matches UUIDs for filtering purposes
+    static ref PING_FILE_NAME_PATTERN: Regex = Regex::new("[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}").unwrap();
+}
+
+/// Attempt to delete a file.
+///
+/// This won't panic if not able to delete, just log.
+fn delete_file(path: &Path) {
+    match remove_file(path) {
+        Err(e) => log::error!("Error deleting file {}. {}", path.display(), e),
+        _ => log::info!("Deleted file {}", path.display()),
+    };
+}
+
+/// Get the file name from a path as a &str.
+///
+/// This won't panic if not able to get file name, just log.
+fn get_file_name_as_str(path: &Path) -> Option<&str> {
+    match path.file_name() {
+        None => {
+            log::warn!("Error getting file name from path: {}", path.display());
+            None
+        }
+        Some(file_name) => {
+            let file_name = file_name.to_str();
+            if file_name.is_none() {
+                log::warn!("File name is not valid unicode: {}", path.display());
+            }
+            file_name
+        }
+    }
+}
+
+/// Creates requests for ping upload and process responses from them.
+pub struct PingUploadHelper {
```

Comment:
Bikeshedding: `UploadRequestManager`? `UploadManager`? `UploadScheduler`?

Reply:
I like `UploadManager`. I'll wait and see if more people have opinions on this before changing it :)